File: 6 kyu/Sum of Digits/solution.py
def sum_of_digits(n):
    result = 0
    for digit in list(str(n)):
        result += int(digit)
    return result

def digital_root(n):
    result = sum_of_digits(n)
    while result > 9:
        result = sum_of_digits(result)
    return result
  
_______________________________________
def digital_root(n):
    return n if n < 10 else digital_root(sum(map(int,str(n))))
  
_______________________________________
def digital_root(n):
    return n%9 or n and 9 
  
_______________________________________
def digital_root(n):
    # ...
    while n>9:
        n=sum(map(int,str(n)))
    return n
  
_______________________________________
def digital_root(n):
    def sm(x):
        return sum(int(i) for i in str(x))
    
    st= sm(n)
    while len(str(st))>1:
        st=sm(st)

    return (st)
    
_______________________________________
def digital_root(n):
    tmp=str(n)
    res=999
    while(len(str(res))!=1):
        res=0
        for i in tmp:
            res+=int(i)
        tmp=str(res)
    return res
  
_______________________________________
def digital_root(n):
    while True:
        s = str(n)
        if len(s) <= 1:
            break
        n = sum(int(x) for x in s)
    return n
  
_______________________________________
def digital_root(n):
    number = list(str(n))
    number = [int(i) for i in number]
    total = sum(number)
    print(total)
    while len(list(str(total))) >= 2:
        number = list(str(total))
        number = [int(i) for i in number]
        total = sum(number) 
        print(total)
    return int(total)

File: 6 kyu/Sum of Digits/solution.cs
public class Number
{
  public int DigitalRoot(long n)
  {
     return (int)(1 + (n - 1) % 9);
  }
}

________________________________
public class Number
{
    public int DigitalRoot(long n)
    {
        if (n < 10) return (int)n;
        long r = 0;
        while (n > 0)
        {
            r += n % 10;
            n /= 10;
        }
        return DigitalRoot(r);
    }
}

________________________________
using System;
using System.Linq;

public class Number
{
  public int DigitalRoot(long n)
  {
    if(n.ToString().Length >1)
      return DigitalRoot(n.ToString().Select(digit=> int.Parse(digit.ToString())).Sum());
    
    return (int)n;
  }
}

________________________________
using System;
using System.Linq;

public class Number
{
  public int DigitalRoot(long n) 
  {
    long count = n.ToString().ToCharArray().Select(x => int.Parse(x.ToString())).Sum();
    
    return count > 9 ? DigitalRoot(count) : Convert.ToInt32(count);
  }
}

________________________________
using System;

public class Number
{
  public int DigitalRoot(long n)
  {
    string num = n.ToString();
    while (num.Length > 1)
    {
      
      int total = 0;
      
      foreach(var c in num)
      {
        total += c - '0';
      }
      
      num = total.ToString();
      
    }
    return Convert.ToInt32(num);
  }
}

________________________________
public class Number
{
  public int DigitalRoot(long n)
        {
            string input = n.ToString();
            var nums = input.ToCharArray();
            int sum = 10;
            while (sum > 9)
            {
                nums = GetSumOfNums(nums, out sum);
            }
            return sum; 
        }
        private char[] GetSumOfNums(char[] nums, out int sum)
        {
            sum = 0;
            foreach (char num in nums)
            {
                sum += int.Parse(num.ToString());
            }
            return sum.ToString().ToCharArray();
        }
}
